### PR TITLE
Replace all JSBin links in audio/video delivery section

### PR DESCRIPTION
**Pull request by Josh-Cena**
Part of https://github.com/mdn/content/issues/16120. Instead of JPlayer, I'm using shared-assets instead due to CSP issues.

### CODE REVIEW

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md [hunk 1]
```diff
@@ -68,41 +70,38 @@ To try out and visualize buffered time ranges we can write a bit of HTML:
 
 and a bit of JavaScript:
 
-```js
-window.onload = () => {
-  const audio = document.getElementById("my-audio");
-  const canvas = document.getElementById("my-canvas");
-  const context = canvas.getContext("2d");
+```js live-sample___timeranges
+const audio = document.getElementById("my-audio");
```
Comment (bsmth) [inline files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:81]: Seems we need a `window.onload = function () {` wrapper for this

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md [hunk 2]
```diff
@@ -55,10 +55,12 @@ audio.buffered.end(1); // returns 19
 
 To try out and visualize buffered time ranges we can write a bit of HTML:
 
-```html
+```html live-sample___timeranges
 <p>
   <audio id="my-audio" controls>
-    <source src="music.mp3" type="audio/mpeg" />
+    <source
+      src="/shared-assets/audio/voices-around-a-fire.mp3"
```
Comment (bsmth) [inline files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md]: The file is a little too short to see buffered segments, so with the `window.onload` wrapper, the loading section goes from grey to red instantly. We can look at that separately, though.

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md [hunk 3]
```diff
@@ -214,8 +212,7 @@ This should give you results similar to the following, where the light grey bar
 
 ![An audio player with play button, seek bar, and volume control, and a progress bar below the controls. The progress bar has a green portion to show played video and a light grey portion to show how much has been buffered.](bufferedprogress.png)
```
Comment (bsmth) [inline files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md]: Do you think we can get rid of the screenshots?

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md [hunk 4]
```diff
@@ -174,36 +175,33 @@ We'll use the following CSS to style the buffering display:
...
+
+audio.addEventListener("progress", () => {
+  const duration = audio.duration;
+  if (duration > 0) {
+    for (let i = 0; i < audio.buffered.length; i++) {
+      if (
+        audio.buffered.start(audio.buffered.length - 1 - i) < audio.currentTime
+      ) {
+        document.getElementById("buffered-amount").style.width = `${
+          (audio.buffered.end(audio.buffered.length - 1 - i) * 100) / duration
```
Comment (bsmth) [inline files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:192]: I don't see the buffered amount (light-grey) in this example, but it works on the jsbin. Played (green, `progress-amount`) looks okay to me, though.

File: files/en-us/web/media/guides/audio_and_video_delivery/index.md
Comment (bsmth) [inline files/en-us/web/media/guides/audio_and_video_delivery/index.md:1]: L75 is missing a period at the end of the sentence if you want to catch it in this PR.

File: files/en-us/web/media/guides/audio_and_video_delivery/webaudio_playbackrate_explained/index.md [hunk 5]
```diff
@@ -39,29 +35,41 @@ Let's create a {{ htmlelement("video") }} element first, and set up video and pl
...
+.holder {
+  width: 640px;
+  height: 360px;
+  margin: 0 auto;
+  margin-bottom: 14em;
+}
+
+#pbr {
+  width: 100%;
+}
```
Comment (bsmth) [inline files/en-us/web/media/guides/audio_and_video_delivery/webaudio_playbackrate_explained/index.md]:
```suggestion
  margin: 0 auto;
  margin-bottom: 14em;
}

form {
  font-family: monospace;
}

video {
  width: 500px;
  aspect-ratio: 16 / 9;
}
```

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:81]: Maybe there's a nicer way to do this, but that's enough to get it working

**Review by bsmth — CHANGES_REQUESTED** [review]
Some things to check, mostly the buffering / seeking examples need another look to get working. Nothing else major.

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md (hunk 1 above)
Comment (Josh-Cena) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:81]: I'm not sure what `onload` changes though. The script is inserted after the DOM so it already has access to the DOM, and even if I wrap it in `onload` I still don't get any red color... which is weird.

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md (hunk 2 above)
Comment (Josh-Cena) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md]: I don't see any red on my end unfortunately. I'm also not sure if I should commit a 5-minute clip to shared-assets hehe. I really wonder if there are external media sources we can trust.

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md (hunk 3 above)
Comment (Josh-Cena) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md]: We could remove them if I can get the buffered area to render

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md (hunk 4 above)
Comment (Josh-Cena) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:192]: Same here, not sure why

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md (hunk 2 above)
Comment (bsmth) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md]: >I'm also not sure if I should commit a 5-minute clip to shared-assets hehe

It's okay to add one longer / larger audio clip to that repo, IMO

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md (hunk 1 above)
Comment (bsmth) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:81]: OK I figured it out, `audio.duration` is not known until the network gets the resource, so we need to wait for `loadedmetadata`:

```js
audio.addEventListener("loadedmetadata", () => {
  const inc = canvas.width / audio.duration;

  // Display TimeRanges

  audio.addEventListener("seeked", () => {
    for (let i = 0; i < audio.buffered.length; i++) {
      const startX = audio.buffered.start(i) * inc;
      const endX = audio.buffered.end(i) * inc;
      const width = endX - startX;

      context.fillRect(startX, 0, width, canvas.height);
      context.rect(startX, 0, width, canvas.height);
      context.stroke();
    }
  });
});
```

Then we don't need `window.onload`.

I tried this with a longer file:

```html
<p>
  <audio id="my-audio" controls>
    <source
      src="https://cdn.freesound.org/previews/687/687928_13113654-lq.mp3"
      type="audio/mpeg" />
  </audio>
</p>
<p>
  <canvas id="my-canvas" width="300" height="20"> </canvas>
</p>
```

And we're looking at the `seeked` event, so you need to click around in the playback position for it to update.

Comment (bsmth) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:81]: Audio comes from https://freesound.org/people/KevinSonger/sounds/687928/

Comment (Josh-Cena) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:81]: Neat, thanks!

Comment (Josh-Cena) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:81]: Do you have a ~10min clip to recommend? Apparently I can use this asset directly on DMO without uploading it to shared-assets, but this 3h one is a bit too long to see the red segments.

Comment (bsmth) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:81]: What about this one? https://freesound.org/people/jgrzinich/sounds/155386/ (`https://cdn.freesound.org/previews/155/155386_326032-lq.mp3`)

Some [others from this search](https://freesound.org/search/?q=&f=license%253A%2522Creative%2520Commons%25200%2522%2520category%253A%2522Soundscapes%2522%2520type%253A%2522mp3%2522&s=Duration+%28longest+first%29&si_tags=0&si_name=0&si_description=0&si_packname=0&si_sound_id=0&si_username=0&d0=0&d1=%2A&ig=0&r=0&g=1&dp=0&cm=0&mm=0&page=18#sound) (CC0)

File: files/en-us/web/media/guides/audio_and_video_delivery/webaudio_playbackrate_explained/index.md [hunk 6]
```diff
@@ -25,12 +25,8 @@ Let's create a {{ htmlelement("video") }} element first, and set up video and pl
 
 ```html
 <video id="myVideo" controls>
```
Comment (bsmth) [inline files/en-us/web/media/guides/audio_and_video_delivery/webaudio_playbackrate_explained/index.md]:
```suggestion
<video id="myVideo" controls loop>
```

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md (hunk 4 above)
Comment (bsmth) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:192]: I think it's because we're using the cached versions so there's nothing to download between test runs

**Review by bsmth — APPROVED** [review]
Thanks a lot, there's a small edge case to cover regarding loading / buffered cached things, but that's not blocking

**Comment by Josh-Cena** [comment]
Yeah idk how to fix that and if we should -_-

File: files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md (hunk 4 above)
Comment (Josh-Cena) [reply files/en-us/web/media/guides/audio_and_video_delivery/buffering_seeking_time_ranges/index.md:192]: It's showing up for me now, so I think it's good
